[Mappings editor] Add support for histogram field type

alisonelizabeth opened the pull request:
Fixes https://github.com/elastic/kibana/issues/63937

I also created a common component for the `meta` parameter. While working on this PR, I found out from the ES team that this parameter is available to most field types so I added support where appropriate via https://github.com/elastic/kibana/pull/76671/commits/df56ddc4e148fd17eb1db28192accf7fd93f30a4.

### Release note
The mappings editor in the Index Templates UI now supports configuring the [histogram field type](https://www.elastic.co/guide/en/elasticsearch/reference/master/histogram.html). Support for the `meta` parameter was also added to the `boolean`, `binary`, `completion`, `date`, `flattened`, `geo_point`, `numeric`, `range`, `search_as_you_type`, `token_count` and `text` field types.

<img width="374" alt="Screen Shot 2020-09-03 at 12 26 36 PM" src="https://user-images.githubusercontent.com/5226211/92144130-3e31d100-ede4-11ea-8c34-0f3b807f896e.png">

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72347/)
* Commit: 80bdc596a9e49b6d5070a38d659461ab5b917880
* This comment will update when the build is complete

### Failed CI Steps
* [Check i18n](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72347/execution/node/394/log)

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72347/","number":"72347","commit":"80bdc596a9e49b6d5070a38d659461ab5b917880"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72363/)
* Commit: fe6cbd5e8c7f6c17104b7cfadd8202d297c38a6f

### Build metrics
#### @kbn/optimizer bundle module count
| id | [value](https://github.com/elastic/kibana/commit/fe6cbd5e8c7f6c17104b7cfadd8202d297c38a6f) | diff | [baseline](https://github.com/elastic/kibana/commit/aab8d3c7ddf97306f00c219f6ab30d6e4cec2618) |
| --- | --- | --- | --- |
| indexManagement | 502 | +2 | 500 |

#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/fe6cbd5e8c7f6c17104b7cfadd8202d297c38a6f) | diff | [baseline](https://github.com/elastic/kibana/commit/aab8d3c7ddf97306f00c219f6ab30d6e4cec2618) |
| --- | --- | --- | --- |
| indexManagement | 1.6MB | +3.6KB | 1.6MB |

### History
* :broken_heart: [Build #72347](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72347/) failed 80bdc596a9e49b6d5070a38d659461ab5b917880

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72347/","number":"72347","commit":"80bdc596a9e49b6d5070a38d659461ab5b917880"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72363/","number":"72363","commit":"fe6cbd5e8c7f6c17104b7cfadd8202d297c38a6f"}]}
PIPELINE-->

elasticmachine commented:
Pinging @elastic/es-ui (Team:Elasticsearch UI)

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72405/)
* Commit: df56ddc4e148fd17eb1db28192accf7fd93f30a4
* This comment will update when the build is complete

### Failed CI Steps
* [Check Types](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72405/execution/node/370/log)

### History
* :green_heart: [Build #72363](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72363/) succeeded fe6cbd5e8c7f6c17104b7cfadd8202d297c38a6f
* :broken_heart: [Build #72347](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72347/) failed 80bdc596a9e49b6d5070a38d659461ab5b917880

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72347/","number":"72347","commit":"80bdc596a9e49b6d5070a38d659461ab5b917880"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72363/","number":"72363","commit":"fe6cbd5e8c7f6c17104b7cfadd8202d297c38a6f"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72405/","number":"72405","commit":"df56ddc4e148fd17eb1db28192accf7fd93f30a4"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72414/)
* Commit: 3e8594084a5f01ab6dd4d3e69461aeae36ca84f5

### Build metrics
#### @kbn/optimizer bundle module count
| id | [value](https://github.com/elastic/kibana/commit/3e8594084a5f01ab6dd4d3e69461aeae36ca84f5) | diff | [baseline](https://github.com/elastic/kibana/commit/aab8d3c7ddf97306f00c219f6ab30d6e4cec2618) |
| --- | --- | --- | --- |
| indexManagement | 502 | +2 | 500 |

#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/3e8594084a5f01ab6dd4d3e69461aeae36ca84f5) | diff | [baseline](https://github.com/elastic/kibana/commit/aab8d3c7ddf97306f00c219f6ab30d6e4cec2618) |
| --- | --- | --- | --- |
| indexManagement | 1.6MB | +5.8KB | 1.6MB |

### History
* :broken_heart: [Build #72405](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72405/) failed df56ddc4e148fd17eb1db28192accf7fd93f30a4
* :green_heart: [Build #72363](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72363/) succeeded fe6cbd5e8c7f6c17104b7cfadd8202d297c38a6f
* :broken_heart: [Build #72347](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72347/) failed 80bdc596a9e49b6d5070a38d659461ab5b917880

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72347/","number":"72347","commit":"80bdc596a9e49b6d5070a38d659461ab5b917880"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72363/","number":"72363","commit":"fe6cbd5e8c7f6c17104b7cfadd8202d297c38a6f"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72405/","number":"72405","commit":"df56ddc4e148fd17eb1db28192accf7fd93f30a4"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72414/","number":"72414","commit":"3e8594084a5f01ab6dd4d3e69461aeae36ca84f5"}]}
PIPELINE-->

sebelga commented:
@elasticmachine merge upstream

sebelga commented on x-pack/plugins/index_management/public/application/components/mappings_editor/components/document_fields/fields/field_types/range_type.tsx:
```diff
@@ -14,11 +14,16 @@ import {
 
-const getDefaultToggleValue = (param: 'locale' | 'format' | 'boost', field: FieldType) => {
+const getDefaultToggleValue = (param: ParameterName, field: FieldType) => {
+  if (param === 'meta') {
```
The default value for meta is `''` so L27 should be enough

sebelga commented on x-pack/plugins/index_management/public/application/components/mappings_editor/components/document_fields/fields/field_types/binary_type.tsx:
```diff
@@ -5,14 +5,24 @@
 
-export const BinaryType = () => {
+const getDefaultToggleValue = (param: ParameterName, field: FieldType) => {
+  return field[param] !== undefined && field[param] !== '';
```
Could we default to the more generic 

```js
return field[param] !== undefined && field[param] !== getFieldConfig(param).defaultValue;
```

like we do in the "range_type"?

sebelga commented on x-pack/plugins/index_management/public/application/components/mappings_editor/components/document_fields/fields/field_types/boolean_type.tsx:
```diff
@@ -24,8 +25,9 @@ const getDefaultToggleValue = (param: string, field: FieldType) => {
     case 'boost': {
       return field[param] !== undefined && field[param] !== getFieldConfig(param).defaultValue;
     }
+    case 'meta':
```
I think "null_value" could have been left, and simply add "meta" on top of "boost"

sebelga commented on x-pack/plugins/index_management/public/application/components/mappings_editor/components/document_fields/fields/field_types/completion_type.tsx:
```diff
@@ -10,14 +10,17 @@ import { i18n } from '@kbn/i18n';
     case 'max_input_length': {
       return field[param] !== undefined && field[param] !== getFieldConfig(param).defaultValue;
     }
+    case 'meta': {
```
I think we can simply add "meta" on top of "max_input_length"

sebelga commented on x-pack/plugins/index_management/public/application/components/mappings_editor/components/document_fields/fields/field_types/date_type.tsx:
```diff
@@ -29,8 +30,9 @@ const getDefaultToggleValue = (param: string, field: FieldType) => {
+    case 'meta':
     case 'null_value': {
-      return field.null_value !== undefined && field.null_value !== '';
+      return field[param] !== undefined && field[param] !== '';
```
Same here: I think "null_value" could have been left, and simply add "meta" on top of "boost"

sebelga commented on x-pack/plugins/index_management/public/application/components/mappings_editor/components/document_fields/fields/field_types/flattened_type.tsx:
```diff
@@ -32,8 +33,9 @@ const getDefaultToggleValue = (param: string, field: FieldType) => {
+    case 'meta':
     case 'null_value': {
-      return field.null_value !== undefined && field.null_value !== '';
+      return field[param] !== undefined && field[param] !== '';
```
Same here :) But on top of "similarity"

sebelga commented on x-pack/plugins/index_management/public/application/components/mappings_editor/components/document_fields/fields/field_types/histogram_type.tsx:
```diff
@@ -0,0 +1,28 @@
+}
+
+const getDefaultToggleValue = (param: ParameterName, field: FieldType) => {
+  return field[param] !== undefined;
```
Can we default to the more generic version?

```js
return field[param] !== undefined && field[param] !== getFieldConfig(param).defaultValue;
```

sebelga commented on x-pack/plugins/index_management/public/application/components/mappings_editor/components/document_fields/fields/field_types/numeric_type.tsx:
```diff
@@ -29,8 +30,9 @@ const getDefaultToggleValue = (param: string, field: FieldType) => {
+    case 'meta':
     case 'null_value': {
-      return field.null_value !== undefined && field.null_value !== '';
+      return field[param] !== undefined && field[param] !== '';
```
Same here :) on top of "ignore_malformed"

sebelga commented on x-pack/plugins/index_management/public/application/components/mappings_editor/components/document_fields/fields/field_types/search_as_you_type.tsx:
```diff
@@ -29,6 +30,9 @@ const getDefaultToggleValue = (param: string, field: FieldType) => {
     case 'max_shingle_size': {
       return field[param] !== undefined && field[param] !== getFieldConfig(param).defaultValue;
     }
+    case 'meta': {
```
same here: on top of "max_shingle_size"

sebelga commented on x-pack/plugins/index_management/public/application/components/mappings_editor/components/document_fields/fields/field_types/token_count_type.tsx:
```diff
@@ -29,8 +30,9 @@ const getDefaultToggleValue = (param: string, field: FieldType) => {
+    case 'meta':
     case 'null_value': {
-      return field.null_value !== undefined && field.null_value !== '';
+      return field[param] !== undefined && field[param] !== '';
```
Same here :)

sebelga reviewed (approved):
Great work @alisonelizabeth ! I left a few comments to tidy things up but no blockers. Cheers.

sebelga commented:
Ups! I went a bit fast on clicking the "Approve button" 😊 
There is a strange bug I discovered.

1. Create a template with an histogram field and a meta like this

```
{ "meta": true }
```

2. Save ---> everything ok
3. Edit the template and the field and set the meta as

```
["meta"] // valid JSON but not valid for ES
```

4. Save ---> error

```
[illegal_argument_exception] composable template [my-template] template after composition is invalid
```

5. Edit the template and put back the initial meta (object) value

6. Save ---> still error?
7. Edit the field... and disable the meta (toggle OFF)
8. Save ---> ok
9. Edit the template + edit the field and add valid meta (object)
10. Save ---> still error??? (this is very strange, we can't save a meta anymore on that field)

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72844/)
* Commit: 7886b0d142d3102c589a38cddeb2e9ebae93c734

### Build metrics
#### @kbn/optimizer bundle module count
| id | [value](https://github.com/elastic/kibana/commit/7886b0d142d3102c589a38cddeb2e9ebae93c734) | diff | [baseline](https://github.com/elastic/kibana/commit/19bf9ed072ca15157e11671745e3783efe27d925) |
| --- | --- | --- | --- |
| indexManagement | 502 | +2 | 500 |

#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/7886b0d142d3102c589a38cddeb2e9ebae93c734) | diff | [baseline](https://github.com/elastic/kibana/commit/19bf9ed072ca15157e11671745e3783efe27d925) |
| --- | --- | --- | --- |
| indexManagement | 1.5MB | +5.8KB | 1.5MB |

### History
* :green_heart: [Build #72414](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72414/) succeeded 3e8594084a5f01ab6dd4d3e69461aeae36ca84f5
* :broken_heart: [Build #72405](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72405/) failed df56ddc4e148fd17eb1db28192accf7fd93f30a4
* :green_heart: [Build #72363](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72363/) succeeded fe6cbd5e8c7f6c17104b7cfadd8202d297c38a6f
* :broken_heart: [Build #72347](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72347/) failed 80bdc596a9e49b6d5070a38d659461ab5b917880

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72347/","number":"72347","commit":"80bdc596a9e49b6d5070a38d659461ab5b917880"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72363/","number":"72363","commit":"fe6cbd5e8c7f6c17104b7cfadd8202d297c38a6f"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72405/","number":"72405","commit":"df56ddc4e148fd17eb1db28192accf7fd93f30a4"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72414/","number":"72414","commit":"3e8594084a5f01ab6dd4d3e69461aeae36ca84f5"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72844/","number":"72844","commit":"7886b0d142d3102c589a38cddeb2e9ebae93c734"}]}
PIPELINE-->

sebelga commented:
[EDIT]

So I played a bit more with it and the steps to reproduce are slightly different:

1. Create an index with JUST a name + and the `logs*` index pattern
2. Save the template (this is the request about to be saved)

<img width="874" alt="Screenshot 2020-09-07 at 11 21 42" src="https://user-images.githubusercontent.com/2854616/92371861-75fa8a80-f0fc-11ea-8445-e0b22212c09d.png">

3. Edit the template and add 1 histogram field. Edit the field to add a meta.

Even without looking at the advanced configuration form (navigate to that tab), we get all of them in the request

<img width="881" alt="Screenshot 2020-09-07 at 11 24 19" src="https://user-images.githubusercontent.com/2854616/92372061-b528db80-f0fc-11ea-9eaf-d261060e8261.png">

And saving it not allowed.

```
[illegal_argument_exception] composable template [my-template] template after composition is invalid
```

4. I then remove the meta (toggle OFF). The request sill has all the advanced settings but I am allowed to save

<img width="882" alt="Screenshot 2020-09-07 at 11 25 21" src="https://user-images.githubusercontent.com/2854616/92372237-ee614b80-f0fc-11ea-9fbb-5064fe6abb4d.png">


In your other PR with constant_keyword, no advanced config is added to the mappings but the same error is returned when adding some `meta` (with `logs*` as index pattern)

<img width="876" alt="Screenshot 2020-09-07 at 12 04 35" src="https://user-images.githubusercontent.com/2854616/92376079-55353380-f102-11ea-9636-334b91613ec6.png">


<img width="860" alt="Screenshot 2020-09-07 at 12 05 03" src="https://user-images.githubusercontent.com/2854616/92376110-61b98c00-f102-11ea-852b-c63aeb4deb8e.png">

alisonelizabeth commented:
Thanks for the review @sebelga! I believe I addressed all of your comments. Would you mind taking another look?

Also, as mentioned in https://github.com/elastic/kibana/pull/76564, I believe the error you were hitting when saving `meta` was due to setting a boolean value instead of a string.

I was not able to reproduce the issue you saw related to the advanced settings config. Can you retest and let me know if you still are able to reproduce?

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74220/)
* Commit: 45363e531ca7f4038cb9460a6c685f899b3c4881

### Build metrics
#### @kbn/optimizer bundle module count
| id | [value](https://github.com/elastic/kibana/commit/45363e531ca7f4038cb9460a6c685f899b3c4881) | diff | [baseline](https://github.com/elastic/kibana/commit/1a49c4e20376cc53ec33acb3d750cf0b943c5cd7) |
| --- | --- | --- | --- |
| indexManagement | 505 | +2 | 503 |

#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/45363e531ca7f4038cb9460a6c685f899b3c4881) | diff | [baseline](https://github.com/elastic/kibana/commit/1a49c4e20376cc53ec33acb3d750cf0b943c5cd7) |
| --- | --- | --- | --- |
| indexManagement | 1.6MB | +6.1KB | 1.5MB |

### History
* :green_heart: [Build #72844](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72844/) succeeded 7886b0d142d3102c589a38cddeb2e9ebae93c734
* :green_heart: [Build #72414](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72414/) succeeded 3e8594084a5f01ab6dd4d3e69461aeae36ca84f5
* :broken_heart: [Build #72405](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72405/) failed df56ddc4e148fd17eb1db28192accf7fd93f30a4
* :green_heart: [Build #72363](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72363/) succeeded fe6cbd5e8c7f6c17104b7cfadd8202d297c38a6f
* :broken_heart: [Build #72347](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72347/) failed 80bdc596a9e49b6d5070a38d659461ab5b917880

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72347/","number":"72347","commit":"80bdc596a9e49b6d5070a38d659461ab5b917880"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72363/","number":"72363","commit":"fe6cbd5e8c7f6c17104b7cfadd8202d297c38a6f"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72405/","number":"72405","commit":"df56ddc4e148fd17eb1db28192accf7fd93f30a4"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72414/","number":"72414","commit":"3e8594084a5f01ab6dd4d3e69461aeae36ca84f5"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72844/","number":"72844","commit":"7886b0d142d3102c589a38cddeb2e9ebae93c734"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74220/","number":"74220","commit":"45363e531ca7f4038cb9460a6c685f899b3c4881"}]}
PIPELINE-->

alisonelizabeth commented:
@elasticmachine merge upstream

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74919/)
* Commit: a24d8a8d0556524fa8ef8177f77f270431c8f341
* This comment will update when the build is complete

### History
* :green_heart: [Build #74220](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74220/) succeeded 45363e531ca7f4038cb9460a6c685f899b3c4881
* :green_heart: [Build #72844](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72844/) succeeded 7886b0d142d3102c589a38cddeb2e9ebae93c734
* :green_heart: [Build #72414](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72414/) succeeded 3e8594084a5f01ab6dd4d3e69461aeae36ca84f5
* :broken_heart: [Build #72405](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72405/) failed df56ddc4e148fd17eb1db28192accf7fd93f30a4
* :green_heart: [Build #72363](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72363/) succeeded fe6cbd5e8c7f6c17104b7cfadd8202d297c38a6f

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72363/","number":"72363","commit":"fe6cbd5e8c7f6c17104b7cfadd8202d297c38a6f"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72405/","number":"72405","commit":"df56ddc4e148fd17eb1db28192accf7fd93f30a4"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72414/","number":"72414","commit":"3e8594084a5f01ab6dd4d3e69461aeae36ca84f5"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72844/","number":"72844","commit":"7886b0d142d3102c589a38cddeb2e9ebae93c734"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74220/","number":"74220","commit":"45363e531ca7f4038cb9460a6c685f899b3c4881"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74919/","number":"74919","commit":"a24d8a8d0556524fa8ef8177f77f270431c8f341"}]}
PIPELINE-->

sebelga reviewed (approved):
Thanks for making the changes! Tested locally and works as expected. You didn't add the validation for the meta values (string) but I guess it will get merged with the other PR.

[Edit] I could not reproduce the issue I mentioned with the advanced settings, which is good 👍

sebelga commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/75133/)
* Commit: ed3889119caa2cfce6f7d8ddf1d09ede95552c8a

### Build metrics
#### @kbn/optimizer bundle module count
| id | [value](https://github.com/elastic/kibana/commit/ed3889119caa2cfce6f7d8ddf1d09ede95552c8a) | diff | [baseline](https://github.com/elastic/kibana/commit/d1a6e89b466e34032bab571cada2c2840fbcd5d7) |
| --- | --- | --- | --- |
| indexManagement | 505 | +2 | 503 |

#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/ed3889119caa2cfce6f7d8ddf1d09ede95552c8a) | diff | [baseline](https://github.com/elastic/kibana/commit/d1a6e89b466e34032bab571cada2c2840fbcd5d7) |
| --- | --- | --- | --- |
| indexManagement | 1.6MB | +6.1KB | 1.5MB |

### History
* :broken_heart: [Build #74919](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74919/) failed a24d8a8d0556524fa8ef8177f77f270431c8f341
* :green_heart: [Build #74220](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74220/) succeeded 45363e531ca7f4038cb9460a6c685f899b3c4881
* :green_heart: [Build #72844](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72844/) succeeded 7886b0d142d3102c589a38cddeb2e9ebae93c734
* :green_heart: [Build #72414](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72414/) succeeded 3e8594084a5f01ab6dd4d3e69461aeae36ca84f5
* :broken_heart: [Build #72405](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72405/) failed df56ddc4e148fd17eb1db28192accf7fd93f30a4

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72405/","number":"72405","commit":"df56ddc4e148fd17eb1db28192accf7fd93f30a4"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72414/","number":"72414","commit":"3e8594084a5f01ab6dd4d3e69461aeae36ca84f5"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72844/","number":"72844","commit":"7886b0d142d3102c589a38cddeb2e9ebae93c734"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74220/","number":"74220","commit":"45363e531ca7f4038cb9460a6c685f899b3c4881"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74919/","number":"74919","commit":"a24d8a8d0556524fa8ef8177f77f270431c8f341"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/75133/","number":"75133","commit":"ed3889119caa2cfce6f7d8ddf1d09ede95552c8a"}]}
PIPELINE-->

alisonelizabeth commented:
Thanks for the second review @sebelga!

> Thanks for making the changes! Tested locally and works as expected. You didn't add the validation for the meta values (string) but I guess it will get merged with the other PR.

The validation is there now that https://github.com/elastic/kibana/pull/76564 has been merged. I also updated the `constant_keyword` field type to use the common `meta` component.

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/75240/)
* Commit: 7974df4d23ee93255df4f4383c5d24c8a38287d6

### Build metrics
#### @kbn/optimizer bundle module count
| id | [value](https://github.com/elastic/kibana/commit/7974df4d23ee93255df4f4383c5d24c8a38287d6) | diff | [baseline](https://github.com/elastic/kibana/commit/9ac2fdfe261cbbbee14f4fbe00aa4de47cefd1bc) |
| --- | --- | --- | --- |
| indexManagement | 506 | +2 | 504 |

#### async chunks size
| id | [value](https://github.com/elastic/kibana/commit/7974df4d23ee93255df4f4383c5d24c8a38287d6) | diff | [baseline](https://github.com/elastic/kibana/commit/9ac2fdfe261cbbbee14f4fbe00aa4de47cefd1bc) |
| --- | --- | --- | --- |
| indexManagement | 1.6MB | +3.7KB | 1.6MB |

### History
* :green_heart: [Build #75133](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/75133/) succeeded ed3889119caa2cfce6f7d8ddf1d09ede95552c8a
* :broken_heart: [Build #74919](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74919/) failed a24d8a8d0556524fa8ef8177f77f270431c8f341
* :green_heart: [Build #74220](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74220/) succeeded 45363e531ca7f4038cb9460a6c685f899b3c4881
* :green_heart: [Build #72844](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72844/) succeeded 7886b0d142d3102c589a38cddeb2e9ebae93c734
* :green_heart: [Build #72414](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72414/) succeeded 3e8594084a5f01ab6dd4d3e69461aeae36ca84f5

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72414/","number":"72414","commit":"3e8594084a5f01ab6dd4d3e69461aeae36ca84f5"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/72844/","number":"72844","commit":"7886b0d142d3102c589a38cddeb2e9ebae93c734"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74220/","number":"74220","commit":"45363e531ca7f4038cb9460a6c685f899b3c4881"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/74919/","number":"74919","commit":"a24d8a8d0556524fa8ef8177f77f270431c8f341"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/75133/","number":"75133","commit":"ed3889119caa2cfce6f7d8ddf1d09ede95552c8a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/75240/","number":"75240","commit":"7974df4d23ee93255df4f4383c5d24c8a38287d6"}]}
PIPELINE-->
